avaframe/com1DFA/particleInitialisation.py

```diff
@@ -0,0 +1,246 @@
+    particles['mIni'] = np.zeros(particles['nPart'])
+    for indRelx, indRely in zip(indRelX, indRelY):
+        # compute number of particles for this cell
+        hCell = relRaster[indRely, indRelx]
+        volCell = dem['areaRaster'][indRely, indRelx] * hCell
+        massCell = volCell * rho
+        ic = indRelx + ncols * indRely
+        iStart = indPartInCell[ic]
+        iStop = indPartInCell[ic+1]
+        indParts = partInCell[iStart:iStop]
+        mNew = massCell / len(indParts)
+        particles['mIni'][indParts] = mNew
+
+    return particles
+
+
+def createReleaseBuffer(cfg, inputSimLines):
+    """ add a buffer around release polygons to get boundary particles
+
+        Parameters
+        -----------
+        cfg: configparser object
+            configuration settings
+        inputSimLines: dict
+            dictionary with input data info
+
+        Returns
+        --------
+        inputSimLines: dict
+            updated inputSimLines with releaseLineBuffer
+    """
+
+    # get start indices and lengths of release polygons
+    lengthRels = inputSimLines['releaseLine']['Length']
+    startLines = inputSimLines['releaseLine']['Start']
+    count = 0
+    xBuffered = np.empty(0)
+    yBuffered = np.empty(0)
+    lengthArray = []
+    startArray = [int(startLines[0])]
+    for m in range(len(lengthRels)):
+
+        # get coordinates of release line for each feature
+        indStart = int(startLines[count])
+        indStop = int(startLines[count] + lengthRels[count])
+        xRel = inputSimLines['releaseLine']['x'][indStart:indStop]
+        yRel = inputSimLines['releaseLine']['y'][indStart:indStop]
+        # create list of point tuples
+        points = []
+        for m in range(len(xRel)):
+            pointsT = xRel[m], yRel[m]
+            points.append(pointsT)
+
+        # create polygon
+        pol1 = Polygon(points)
+        # compute bufferZone
+        bufferZone = cfg['GENERAL'].getfloat('sphKernelRadius') * cfg['GENERAL'].getfloat('bufferZoneFactor')
+        polBuffered = pol1.buffer(bufferZone)
+        xnew, ynew = polBuffered.exterior.coords.xy
+        xnew = np.asarray(xnew)
+        ynew = np.asarray(ynew)
+        xBuffered = np.append(xBuffered, xnew)
+        yBuffered = np.append(yBuffered, ynew)
+        lengthArray.append(len(xnew))
+        startArray.append(int(startArray[-1]+len(xnew)))
+        count = count + 1
+
+    # add buffered release lines to dictionary
+    releaseLineBuffer = inputSimLines['releaseLine'].copy()
+    releaseLineBuffer['x'] = xBuffered
+    releaseLineBuffer['y'] = yBuffered
+    releaseLineBuffer['Length'] = np.asarray(lengthArray)
+    releaseLineBuffer['Start'] = np.asarray(startArray)
+    releaseLineBuffer['z'] = np.zeros(len(xBuffered))
+    inputSimLines['releaseLineBuffer'] = releaseLineBuffer
+
+    # plt.plot(inputSimLines['releaseLine']['x'], inputSimLines['releaseLine']['y'], 'g')
+    # plt.plot(xBuffered, yBuffered, 'b')
+    # plt.show()
```

Comment:
please remove or put behind an existing plot flag

Reply:
moved to debugPlots